subsys/bluetooth/host/gatt.c

```diff
@@ -2110,47 +2078,74 @@ static bool gatt_cf_notify_multi(struct bt_conn *conn)
 	return CF_NOTIFY_MULTI(cfg);
 }
 
-static int gatt_notify_mult(struct bt_conn *conn, uint16_t handle,
+static int gatt_notify_mult(struct bt_conn *conn, uint16_t num_params,
 			    struct bt_gatt_notify_params *params)
 {
-	struct net_buf **buf = &nfy_mult[bt_conn_index(conn)];
+	struct net_buf *buf;
 	struct bt_att_notify_mult *nfy;
+	struct notify_data data;
+	uint16_t pdu_len = 0;
+
+	for (uint16_t i = 0; i < num_params; i++) {
+		data.attr = params[i].attr;
+		data.handle = bt_gatt_attr_get_handle(params[i].attr);
+		/* Lookup UUID if it was given */
+		if (params[i].uuid) {
+			if (!gatt_find_by_uuid(&data, params[i].uuid)) {
+				return -ENOENT;
+			}
+			params[i].attr = data.attr;
```

Comment:
I know this is part of the existing solution, but I highly dislike updating the parameters like this (but no use fixing it just for notify_mult).

Reply:
Yeah I recently encountered it in the gatt subscribed check, it was pretty misleading in gdb. I would be o.k. with either having a commit fixing this in the whole of gatt.c or merging it this way and then making a subsequent PR that fixes this kind of behavior.
I can draft that PR if you don't have the time @laxiLang , just tell me.

Reply:
@jori-nordic Unfortunately I think this is a general issue in GATT. It's quite typical for GATT to modify the data in the parameter structs (for better or worse). 

It will be a significant API change to fix, so at this point the best course of action might be to add additional documentation about it.

Reply:
@jori-nordic if you help with the proposal for a fix, do appreciate it.

Reply:
@Thalley would you be ok merging this as-is (I mean this comment specifically) if I create a doc PR right after ? I don't want to add more noise to that one.

Reply:
Yes, that is fine